homeassistant/components/camera/__init__.py

```diff
@@ -578,17 +584,32 @@ async def stream_source(self) -> str | None:
         """
         return None
 
-    async def async_handle_web_rtc_offer(self, offer_sdp: str) -> str | None:
+    async def async_handle_web_rtc_offer(
+        self, offer_sdp: str, session_id: str, send_message: WebRTCSendMessage
+    ) -> None:
         """Handle the WebRTC offer and return an answer.
 
         This is used by cameras with CameraEntityFeature.STREAM
         and StreamType.WEB_RTC.
 
         Integrations can override with a native WebRTC implementation.
         """
-        for provider in self._webrtc_providers:
-            if answer := await provider.async_handle_web_rtc_offer(self, offer_sdp):
-                return answer
+        if (
+            self._webrtc_provider
+            and await self._webrtc_provider.async_handle_web_rtc_offer(
+                self, offer_sdp, session_id, send_message
+            )
+        ):
+            return
+
+        if self._legacy_webrtc_provider and (
+            answer := await self._legacy_webrtc_provider.async_handle_web_rtc_offer(
```

Comment:
Can't we migrate all legacy webrtc providers to just call `send_message` ?

Reply:
No as they support only the sync approach, but I recommend to deprecate them in a follow up PR

Reply:
We can't just do this?
https://github.com/allenporter/home-assistant-core/commit/d65c8994eba8beb044892f46f52d6a576b7bc43e